assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -392,11 +394,36 @@ const SelectionDataLayers = ({
   const routePatterns: ByRoutePatternId<RoutePattern> | null =
     usePatternsByIdForRoute(routePatternIdentifier?.routeId || null)
 
-  const selectVehicle = (vehicleOrGhost: Vehicle | Ghost) =>
-    setSelection({
-      type: SelectedEntityType.Vehicle,
-      vehicleId: vehicleOrGhost.id,
-    })
+  const streetViewActive = useContext(StreetViewModeEnabledContext)
+
+  const selectVehicle: (vehicleOrGhost: Vehicle | Ghost) => void =
+    !streetViewActive
+      ? (vehicleOrGhost) => {
+          setSelection({
+            type: SelectedEntityType.Vehicle,
+            vehicleId: vehicleOrGhost.id,
+          })
+        }
+      : (vehicleOrGhost) => {
+          if (isVehicle(vehicleOrGhost)) {
+            const url = streetViewUrl({
+              latitude: vehicleOrGhost.latitude,
+              longitude: vehicleOrGhost.longitude,
+              bearing: vehicleOrGhost.bearing,
+            })
+
+            window.FS?.event("User clicked map vehicle to open street view", {
+              streetViewUrl_str: url,
+              clickedMapAt: {
+                latitude_real: vehicleOrGhost.latitude,
+                longitude_real: vehicleOrGhost.longitude,
+                bearing_real: vehicleOrGhost.bearing,
+              },
+            })
+
+            window.open(url, "_blank")
+          }
+        }
```

Comment:
**thought:** I know it's already on your TODO list (albeit with a question mark), but I want to flag that I think this and the corresponding changes in `mapMarkers.tsx` would be really good to have unit test coverage for given that they're both kind of complicated logic.

Reply:
I was looking for feedback around where the best places to add tests for this would be, and figured I'd seed it with the ideas I already had, so thanks for confirming! Will see what I can add.

Reply:
Tests pushed up! See [`fc4cde4` (#2117)](https://github.com/mbta/skate/pull/2117/commits/fc4cde46c99c95d56bb8ff60025a4b8e0143c118) for the specific commit.